wso2/analytics-apim
# Fix for #1023

## Purpose
Resolves https://github.com/wso2/analytics-apim/issues/1023

## Goals
When a custom theme is enabled via deployment.yaml, but a given tenant has not added the custom theme (logo and favicon images ) for that tenant themes get fall backs to default wso2 images.

## Approach
`themeConfigResourcesPath` is no longer read from the deployment.yaml. Instead it is considered to be the HOME/wso2/dashboard/deployment/web-ui-apps/analytics-dashboard/public/ directory.

Any tenant who wishes to customize has to create a directory with the tenant name in the above location and include the following files.

<img width="191" alt="theme" src="https://user-images.githubusercontent.com/3957104/75748415-69f4d100-5d45-11ea-8d35-e843040c6885.png">


Base URL and app context to fetch the file is retrieved from the deployment.yaml

*Steps to verify*
1. Add following configuration in the deployment.yaml
`themeConfigProviderClass: org.wso2.analytics.apim.dashboards.theme.config.provider.CustomDashboardThemeConfigProvider`
`logoFileName: download.png`
`faviconFileName: favicon.ico`
2. Create a tenant in APIM node ( say abc.com)
3. Go to HOME/wso2/dashboard/deployment/web-ui-apps/analytics-dashboard/public/  directory and create directory with the name of tenant ( abc.com)
4. Create following folder structure inside the tenant folder and add relavent logo and favicon file.

<img width="191" alt="theme" src="https://user-images.githubusercontent.com/3957104/75748424-70834880-5d45-11ea-8993-665184bb956b.png">


5. Login with the abc.com tenant and favicon and logo should be changed to the added custom images.
6. Login with super tenant admin and default wso2 logo's should be shown.
